prettier/main.ts

```diff
@@ -26,6 +26,7 @@ Options:
   -H, --help                 Show this help message and exit.
   --check                    Check if the source files are formatted.
   --write                    Whether to write to the file, otherwise it will output to stdout, Defaults to false.
+  --from-stdin               Read the typescript/javascript code from stdin and output formatted code to stdout.
```

Comment:
Shouldn’t you be able to guess if something is on stdin? Like if there are no other arguments ?

Reply:
This seems unlikely.

Suppose we judge by condition whether it wants to read the stream from the pipeline.

```
if (stdinIsEmpty) {
  // read code from stdin and format
} else {
  // format the file
}
```

The key is that we can't tell if stdin is empty.

another solution is: create and another file. eg. `prettier/pipe.ts`

then we do this.

```bash
cat test.ts | deno run prettier/pipe.ts
```

Reply:
How about we default to reading from `stdin` if there are no files passed as arguments? This still has one caveat that prettier will hang if there's nothing to read from `stdin` - maybe some hacky solution with `Promise.race` and timeout?

Reply:
Currently, if there is no input file parameter, it traverses the current working directory.

https://github.com/denoland/deno_std/blob/4317af11d18e67dbc79b0df9839180f8e4e55613/prettier/main.ts#L324-L330

Reply:
Can you use ` Deno.isTTY().stdin` ?

Reply:
`Deno.isTTY().stdin` works for me. 👍
I will add test cases later.